src/utils/cmds/putval.c

```diff
@@ -49,10 +58,37 @@ static int set_option(value_list_t *vl, const char *key, const char *value) {
 
     if ((errno == 0) && (endptr != NULL) && (endptr != value) && (tmp > 0.0))
       vl->interval = DOUBLE_TO_CDTIME_T(tmp);
-  } else
-    return 1;
+  } else if (strncasecmp("meta:", key, 5) == 0) {
+    const char *meta_key = key + 5;
+    size_t value_len = strlen(value);
+
+    if (vl->meta == NULL) {
+      vl->meta = meta_data_create();
+      if (vl->meta == NULL) {
+        return CMD_ERROR;
+      }
+    }
 
-  return 0;
+    if (is_quoted(value, value_len)) {
+      int metadata_err;
+      const char *value_str = sstrndup(value + 1, value_len - 2);
```

Comment:
What the ... are you doing here?

Instead of inventing square wheels, you really need: 

```
-      const char *value_str = sstrndup(value + 1, value_len - 2);
+      char *value_str = strdup(value + 1);
+      value_str[value_len - 2] = '\0';
```

I'm chocked seeing ` exit(3);` in `sstrndup` implementation.

Reply:
So you are probably as shocked to see `exit(3)` in the `sstrdup` implementation from Octo three years ago...

Reply:
That is not reason to copy errors.

Reply:
Accidentally found comment from @octo in https://github.com/collectd/collectd/pull/2352

>That doesn't make sense, since the difference between strdup() and sstrdup() 
>is that the latter exits. I'd prefer to switch to strdup() here 
>(and leave the error handling). 
>Unfortunately I haven't found the energy to remove the uses of sstrdup() globally yet.

This only confirms my vision,  what no `exit()` should be called in _monitoring_ system in case of OOM.